tree/ntuple/v7/src/RField.cxx

```diff
@@ -1136,8 +1136,12 @@ void ROOT::Experimental::RFieldBase::ConnectPageSource(Internal::RPageSource &pa
             fColumnRepresentative = &t;
       }
       R__ASSERT(fColumnRepresentative);
-      if (fOnDiskId != kInvalidDescriptorId)
-         fOnDiskTypeVersion = desc.GetFieldDescriptor(fOnDiskId).GetTypeVersion();
+      if (fOnDiskId != kInvalidDescriptorId) {
+         const auto &fieldDesc = desc.GetFieldDescriptor(fOnDiskId);
+         fOnDiskTypeVersion = fieldDesc.GetTypeVersion();
+         if (fieldDesc.GetStreamerChecksum().has_value())
+            fOnDiskStreamerChecksum = fieldDesc.GetStreamerChecksum().value();
```

Comment:
Since we're explicitly checking for `has_value()`, we should use `*fieldDesc.GetStreamerChecksum()` (unchecked access) rather than `value()` here